libs/libvtrutil/src/vtr_logic.h

```diff
@@ -1,5 +1,15 @@
+// Put this above guard so that TRUE/FALSE are undef'ed
+// even if this file was already included earlier.
+#ifdef FALSE
+#    undef FALSE
+#endif
+
+#ifdef TRUE
+#    undef TRUE
+#endif
```

Comment:
Is there a potential issue with code including this header but relying on TRUE/FALSE to be defined (e.g. other header files included after this one)?

The TRUE/FALSE in the enum are scoped (vtr::LogicValue::TRUE/FALSE), but any usage of TRUE/FALSE defines may not be. Thoughts?

Reply:
The best way to resolve this is to rename `vtr::LogicValue::TRUE/FALSE`, because these are defined as [macros on MacOS](https://opensource.apple.com/source/xnu/xnu-6153.11.26/osfmk/mach/boolean.h.auto.html).

I found a workaround defining the macros to themselves, preventing redefinition by `boolean.h`, and then defining constexprs for `TRUE` and `FALSE`.